### ISSUE TITLE
CSV with commas

### ISSUE THREAD

**Issue by pasimme** [JoshClose/CsvHelper#966]
Hello! 

I got a CSV from a client, created in Excel.  When I open in the Mac TextEditor, I see that Excel saved it with double quotes around the field that contains commas; no other fields contain commas nor double quotes:

```
Description,ManufacturerPartNumber,PackageType,Status,Manufacturer,Distributor,DistributorPartNumber,DistributorPackageType,StorageLocation,InventoryId,OnHand,OnOrder
"C-X7R,1F,0603,50aa,10,-55 C~125 C",aaa,CutTape,Active,66,3,bbb,CutTape,Notebook,0,0,0
```

The file contains headers; the only data row starts with "C-X7r...".   All fields are strings, except the last three (InventoryId,OnHand,OnOrder). 

Here is the class/POCO I'm trying to map the CSV data to:
```
public class PartsCsvHeaders
{
    public string Description { get; set; }
    public string ManufacturerPartNumber { get; set; }
    public string PackageType { get; set; }
    public string Status { get; set; }
    public string Manufacturer { get; set; }
    public string Distributor { get; set; }
    public string DistributorPartNumber { get; set; }
    public string DistributorPackageType { get; set; }
    public string StorageLocation { get; set; }
    public int? InventoryId { get; set; }
    public int? OnHand { get; set; }
    public int? OnOrder { get; set; }
}
```

The error I'm getting is: 
> The conversion cannot be performed. Text: 'bbb' MemberType: System.Nullable`1[[System.Int32, System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]] TypeConverter: 'CsvHelper.TypeConversion.NullableConverter'

It seems as though 'bbb' is being thought of as InventoryId, which is an int, when it should be mapped to DistributorPartNumber (string). 

What's going on with this parsing?  Is my error because only the first record is surrounded by double quotes but all others are not? 

Thanks, 
Dan

**Comment by pasimme** [comment]
Here's some more code I'm using to call CsvReader:
```
using (var reader = new StreamReader(file.OpenReadStream(), Encoding.GetEncoding("iso-8859-1")))
{
    var bad = new List<string>();
    var csv = new CsvReader(reader);
    csv.Configuration.BadDataFound = context =>
    {
        bad.Add(context.RawRecord);
    };
    var partsCsv = csv.GetRecords<PartsCsvHeaders>().ToList();
}
```

The exception is thrown when the 'var partsCsv...' line is executed.

**Comment by JoshClose** [comment]
Does the file have a line ending of only `\r`? There is a bug right now where certain files that use `\r` line endings instead of `\r\n` or `\n` don't work properly.

**Comment by pasimme** [comment]
I think I found my issue.  /r are not valid for line break/carriage return (from some researching I did).  Excel for Mac2011 (yes, it's old, but seems to work) was saving with line breaks and special characters (like a degree symbol) with odd encoding, I'm guessing special for a Mac.  I could see that the original CSV used plain line breaks when inspecting/opening the file with Terminals, but when Excel saved it, the line breaks looked like ^M  (instead of plain old linebreaks visible in Terminals).

I downloaded a Mac app called Table Tool that simply saves in UTF-8, and that fixed my issue, since I can edit the CSV and save it in the proper encoding. 

Thanks again!  Hope this post helps people in the future.
